src/sentry/migrations/0186_add_externalactor.py

```diff
@@ -0,0 +1,93 @@
+# Generated by Django 1.11.29 on 2021-04-13 22:38
+
+import django.db.models.deletion
+import django.utils.timezone
+from django.db import migrations, models
+
+import sentry.db.models.fields.bounded
+import sentry.db.models.fields.foreignkey
+import sentry.types.integrations
+
+
+class Migration(migrations.Migration):
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not.
+    # By default we prefer to run in a transaction, but for migrations where you want
+    # to `CREATE INDEX CONCURRENTLY` this needs to be set to False. Typically you'll
+    # want to create an index concurrently when adding one to an existing table.
+    # You'll also usually want to set this to `False` if you're writing a data
+    # migration, since we don't want the entire migration to run in one long-running
+    # transaction.
+    atomic = True
+
+    dependencies = [
+        ("sentry", "0185_rm_copied_useroptions"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="ExternalActor",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("date_updated", models.DateTimeField(default=django.utils.timezone.now)),
+                ("date_added", models.DateTimeField(default=django.utils.timezone.now, null=True)),
+                (
+                    "provider",
+                    sentry.db.models.fields.bounded.BoundedPositiveIntegerField(
+                        choices=[
+                            (sentry.types.integrations.ExternalProviders(100), "email"),
```

Comment:
You should paste a copy of the enum at the top of the file. That way this migration has no dependencies to the repo.

Reply:
This is kind of a pain:
 - If we manually modify the migration to include the enum and use that here then (afaik) `makemigrations` will detect changes, since the enums aren't the same thing, so we can't really do it, unless someone knows of a workaround.
 - If we don't, then whenever we modify the enum it will modify all migrations that have ever used that enum.

I wonder if it's worth investigating https://pypi.org/project/django-enumfield/. We'd have to check whether this solves our problem.